zebrad/src/components/inbound.rs

```diff
@@ -262,8 +268,11 @@ impl Service<zn::Request> for Inbound {
 
                     // Send a sanitized response
                     let mut peers = peers.sanitized();
-                    const MAX_ADDR: usize = 1000; // bitcoin protocol constant
-                    peers.truncate(MAX_ADDR);
+
+                    // Truncate the list
+                    let truncate_at = MAX_ADDR
+                        .min((peers.len() as f64 * FRAC_OF_AVAILABLE_ADDRESS).ceil() as usize);
```

Comment:
Note: We usually try to avoid floating-point calculations in security-critical code, because they can have unexpected precision. (For example, `2^53 + 1 as f64` equals `2^53`.) 

But the numbers here will never be that large, and the consequences of any rounding are a small change in the number of peers. So it's fine here.

We don't need to make any changes.